version: "3.8"

services:
  grafana:
    image: grafana/grafana-oss:latest
    volumes:
      - grafana-data:/var/lib/grafana
    deploy: 
      mode: replicated
      replicas: 1 
      restart_policy: 
        condition: on-failure
      placement:
        constraints: [node.role == manager]
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=traefik"
        - "traefik.http.routers.grafana.rule=Host(`grafana.localhost`)"
        - "traefik.http.routers.grafana.entrypoints=web"
        - "traefik.http.routers.grafana.service=grafana"
        - "traefik.http.services.grafana.loadbalancer.server.port=3000"
    networks:
      - monitoring
      - traefik

volumes:
  grafana-data:
    driver: local

networks:
  traefik:
    external: true
  monitoring:
    external: true
